### ISSUE TITLE
Cleanup pkg/registry directory

### ISSUE THREAD

**Issue by wojtek-t** [kubernetes/kubernetes#12207]
After #11878 and #12020 pkg/registry is no longer coupled with etcd - it is now (relatively) generic.

Few thing that we should cleanup there:
- [ ] move any references to etcd to something more generic (e.g. pkg/registry/*/etcd to pkg/registry/*/storageclient) - please suggest better name for it :)

- [x] migrate services to use rest.StandardStorage and move it to service/ directory (from pkg/registry/etcd/)

- [x] unify limitrange and events with all other resources (all other resources have: registry.go, rest.go and etcd/etcd.go files) - events and limitrange are structured differently

- [x] refactor tests under pkg/registry/*/etcd directoris to use common testing infrastructure from pkg/api/rest/resttest

cc @thockin @smarterclayton @derekwaynecarr @lavalamp @nikhiljindal @fgrzadkowski @davidopp

**Comment by wojtek-t** [comment]
We should also refactor tests under etcd/ directories - all of them are pretty similar and there is huuuuge code duplication there. We already have some "helper" test methods under:
pkg/api/rest/resttest.
We should just extend them and use standard one wherever it is possible.

This would be extremely helpful when we will need to switch to the new etcd client - see #11962, which unfortunately has a different API - changing all those tests then will be a disaster then.

**Comment by bgrant0607** [comment]
One of my biggest gripes is business logic in pkg/registry/foo/etcd/etcd.go, such as setPodHostAndAnnotations. 

More generally, there's no rationale for how our API-related code is spread all over the source tree.

See also #8190, #4851, and #1957.

**Comment by wojtek-t** [comment]
>
One of my biggest gripes is business logic in pkg/registry/foo/etcd/etcd.go, such as setPodHostAndAnnotations.
More generally, there's no rationale for how our API-related code is spread all over the source tree.

Yes - in general I agree. But fixing it requires a lot of time which I don't have now (although I think it would be good to do it).

**Comment by wojtek-t** [comment]
The remaining point is a duplicate of #17546 - closing this one.
